sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/regexpExpressions.scala

```diff
@@ -642,7 +642,7 @@ case class RegExpReplace(subject: Expression, regexp: Expression, rep: Expressio
     }
     val source = s.toString()
     val position = i.asInstanceOf[Int] - 1
-    if (position < source.length) {
+    if (position < source.length || (position == 0 && source.equals(""))) {
```

Comment:
This is minor, but check `source.length == 0` instead? faster.
In fact, isn't this equivalent to `position == 0 || position < source.length`?

Reply:
Yes that's a good point. Compiler actually suggests `source.isEmpty()`. The default position passed in is 1 so position will be 0 by default so `position == 0 || position < source.length` doesn't catch the empty string.
I will replace with `isEmpty`.

I would also argue that any non-default position passed in with an empty string for the regex replace is an error on the user side, so I'm wondering if we should special case that?
As in, if user gives us a regex replace with empty string match (`^$`) and explicitly sets a position, we should probably just throw? I don't really have a strong opinion here, happy to keep as is. Seems just a bit odd to specify a position when you are matching on an empty string that should always have length 0

Reply:
Changed to `isEmpty` in 0e8d78555f0fc54f6ae7331efa743bdb39c30819. Thanks for the look @srowen , let me know what you think :)

Reply:
Hm, how doesn't it catch the empty string? in that case, adding `position == 0 ||` makes the check true iff the string is empty (`position < source.length` is `false`)

Reply:
If position is 0, then the check is currently true in all cases except `source.length == 0` (empty string, right?). Your change makes it true in this case too when position is 0. I think both versions do that?